### PR TITLE
Add docs page for MultiDomainFunction

### PR DESCRIPTION
**Pull request by StephenSmith25**
**Description of work.**
Adds a documentation page for MultiDomainFunction

**To test:**

- Check docs build and the page formats correctly
- Check docs make sense

Contributes to  #16480. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/fitting/fitfunctions/MultiDomainFunction.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
+.. _func-MultiDomainFunction:
+
+=======================
+Multi Domain Function
+=======================
```
Comment (robertapplin) [inline docs/source/fitting/fitfunctions/MultiDomainFunction.rst:5]: I haven't checked the docs build yet, but does this underline need to line up with the text for it to work?

File: docs/source/fitting/fitfunctions/MultiDomainFunction.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+Description
+-----------
+
+A multidomain function includes information about
+which domain a particular entry (function) will be applied to.
+The members of the multidomain function are indexed from 0 to the number of functions minus 1.
+Multidomain functions do not have their own parameters, instead they use
+parameters of the member functions. Parameter names are formed from the
+member function's index and its parameter name: f[domain_index].[name]. For
+example, "f0.A0" would refer to the "A0" parameter of the first function in the multidomain function.
```
Comment (robertapplin) [inline docs/source/fitting/fitfunctions/MultiDomainFunction.rst:18]: Nice description 👍

File: docs/source/fitting/fitfunctions/MultiDomainFunction.rst [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    func = FunctionFactory.Instance().createInitialized(multiFunc)
+    fit_output = Fit(Function=func,
+                     InputWorkspace=ws, WorkspaceIndex=0,
+                     InputWorkspace_1=ws, WorkspaceIndex_1=1,
+                     Output='fit')
+
+As $domains=1 is set for the Quadratic function, it will be used to fit the second domain of the input workspace,
+corresponding to WorkspaceIndex=1. Similarly, the LinearBackground will be used to fit the first domain of the workspace,
+WorkspaceIndex=0. If instead we had specified domains=i, the function would have been fit based on the order of creation,
+i.e. the quadratic function would fit to the first domain and the LinearBackground to the second domain.
```
Comment (robertapplin) [inline docs/source/fitting/fitfunctions/MultiDomainFunction.rst:55]: Nice explanation of `$domains=i`, would it be possible to explain what `$domains=All` means? I am guessing it means the composite is used for all domains?

**Review by robertapplin — CHANGES_REQUESTED** [review]
Very nice additions to this documentation - I will definitely use this myself!

File: docs/source/fitting/fitfunctions/MultiDomainFunction.rst (hunk 1 above)
Comment (StephenSmith25) [reply docs/source/fitting/fitfunctions/MultiDomainFunction.rst]: Not sure to be honest, but I lined it up just to make sure.

File: docs/source/fitting/fitfunctions/MultiDomainFunction.rst (hunk 3 above)
Comment (StephenSmith25) [reply docs/source/fitting/fitfunctions/MultiDomainFunction.rst]: As you say, I think it means its used for all domains, but couldn't find anything describing its behaviour. I tested it out on a simple fit and that’s what it seems to do, so I've added that to the docs.

**Review by robertapplin — APPROVED** [review]

**Pull request merged**